wasd-controls.js

```diff
@@ -34,6 +36,7 @@ WL.registerComponent('wasd-controls', {
         direction[0] *= this.speed;
         direction[2] *= this.speed;
         vec3.transformQuat(direction, direction, this.headObject.transformWorld);
+        if (this.restrictY) direction[1] = 0;
```

Comment:
Unfortunately, this will lead to the classic strafing bug you see in some older games. Instead, try the following:
 - rotate
 - set y to 0
 - normalize, if length not 0 (which might happen if user is looking straight up)
 - multiply by speed